### DISCUSSION TITLE
Running ESP32-TUX using Apache NuttX - TUX-NUT? or NUT-TUX? ;)

### DISCUSSION THREAD

**Discussion by maxbrito500** [sukesh-ak/ESP32-TUX#22] [Ideas]
Hello Sukesh, very good work!

As a template there is already so much done, the only thing missing is to launch different apps either scripted or through a VM.

I'm looking to launch apps using the template format described here: https://github.com/radio3-network/BluemchenOS/blob/main/app-template.md which is similar to OSX where an app is a folder that contains resources for launching either scripts or bytecode binaries.

There is a wide selection of VM and scripted languages for ESP32: https://github.com/radio3-network/operations#virtual-machines-for-esp32

**Question:** What would your recommendations/plans for adding that functionality?

**Comment by sukesh-ak** [comment]
Hi @maxbrito500 
Thank you for the appreciation.
Those are great resources available on that link.

ESP32-TUX was built as a stepping stone for a commercial product of mine.
Currently I am extremely busy probably till end of March.

After that I am planning to build a version which uses Apache NuttX so that it can be ported to more platforms (simulator, SMT32, ESP32-XX, RPI etc...) with little changes. ESP32 is already supported on Apache NuttX though less known for now.

Currently I have got the basics running on Ubuntu (as simulator) with LVGL as well.
https://github.com/sukesh-ak/Apache-NuttX-Getting-Started

**Comment by maxbrito500** [comment]
Thanks Sukesh, those are very good news. I'll keep following your progress and wish you the best for the commercial endeavour.

Was reading on the nuttx features, it seems that binary loading is already available there: https://nuttx.apache.org/docs/latest/components/binfmt.html

This would likely be the path to add new apps on ESP32-TUX that need specific C libraries and can't be scripted, right?

**Comment by sukesh-ak** [comment]
Thanks for the wishes.

Yes binary loading is available and also there is a shell called `nsh` as an example to load other `Apps` already.
`ELF` was the plan but need to spend more time to understand `NXFLAT` and its benefits in comparison.

I haven't yet flashed onto a hardware device yet. So, some research work pending which needs focused time :)
Will rename this thread title with this context.
